### PR TITLE
Aura test update (master)

### PR DESCRIPTION
**Pull request by General-Beck**

### CODE REVIEW

**Comment by arkpar** [comment]
This test should use a temp data dir which should be cleared before the execution

File: scripts/aura-test.sh [hunk 1]
```diff
@@ -1,9 +1,25 @@
 #!/bin/bash
-cargo build -j $(nproc) --release --features final $CARGOFLAGS
+rm -rf parity-import-tests/
+cargo build -j $(nproc) --release --features final
```
Comment (arkpar) [inline scripts/aura-test.sh]: no need for `features final`, this is not a public build

File: scripts/aura-test.sh [hunk 2]
```diff
@@ -1,9 +1,25 @@
 #!/bin/bash
-cargo build -j $(nproc) --release --features final $CARGOFLAGS
+rm -rf parity-import-tests/
+cargo build -j $(nproc) --release --features final 
 git clone https://github.com/paritytech/parity-import-tests
 cp target/release/parity parity-import-tests/aura/parity
```
Comment (arkpar) [inline scripts/aura-test.sh]: no need to copy the binary

File: scripts/aura-test.sh [hunk 3]
```diff
@@ -1,9 +1,23 @@
 #!/bin/bash
-cargo build -j $(nproc) --release --features final $CARGOFLAGS
+rm -rf parity-import-tests/
+cargo build -j $(nproc) --release
 git clone https://github.com/paritytech/parity-import-tests
```
Comment (arkpar) [inline scripts/aura-test.sh]: This should really be a submodule

File: scripts/aura-test.sh [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
-cp target/release/parity parity-import-tests/aura/parity
-cd parity-import-tests/aura
+target/release/parity -v
 echo "Start Aura test"
-parity import blocks.rlp --chain chain.json
-parity restore snap --chain chain.json
+target/release/parity import parity-import-tests/aura/blocks.rlp --chain parity-import-tests/aura/chain.json
```
Comment (arkpar) [inline scripts/aura-test.sh]: -d `/tmp/aura-test-data`

File: scripts/aura-test.sh [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
+if [ $? -eq 0 ]
+then
+  echo "Import test passed"
+else
+  echo "Import test failed" >&2
+  exit 1
+fi
+target/release/parity restore parity-import-tests/aura/snap --chain parity-import-tests/aura/chain.json
```
Comment (arkpar) [inline scripts/aura-test.sh]: `-d /tmp/aura-test-data`

File: scripts/aura-test.sh [hunk 6]
```diff
@@ -1,9 +1,23 @@
 #!/bin/bash
-cargo build -j $(nproc) --release --features final $CARGOFLAGS
+rm -rf parity-import-tests/
```
Comment (arkpar) [inline scripts/aura-test.sh]: Datadir must be cleared as well.
`rm -rf  /tmp/aura-test-data`

**Comment by rphmeier** [comment]
In the future, there will be multiple tests in the `parity-import-tests` repo. After #6085 lands, I will add one in the `tendermint` directory. The future-proof solution would be to iterate over each directory in the repo and perform the import test commands for all of them.

File: scripts/aura-test.sh [hunk 7]
```diff
@@ -1,9 +1,23 @@
 #!/bin/bash
-cargo build -j $(nproc) --release --features final $CARGOFLAGS
-git clone https://github.com/paritytech/parity-import-tests
-cp target/release/parity parity-import-tests/aura/parity
-cd parity-import-tests/aura
+rm -rf parity-import-tests
```
Comment (arkpar) [inline scripts/aura-test.sh]: This is not needed anymore

File: scripts/aura-test.sh [hunk 8]
```diff
@@ -1,9 +1,23 @@
...
+rm -rf /tmp/aura-test-data
+cargo build -j $(nproc) --release
+target/release/parity -v
 echo "Start Aura test"
-parity import blocks.rlp --chain chain.json
-parity restore snap --chain chain.json
+target/release/parity import test/parity-import-tests/aura/blocks.rlp --chain test/parity-import-tests/aura/chain.json -d /tmp/aura-test-data
```
Comment (rphmeier) [inline scripts/aura-test.sh]: @General-Beck there are now multiple import tests. Can you rewrite this as a loop to be future-proof?

Comment (General-Beck) [inline scripts/aura-test.sh]: @rphmeier What condition can be taken for a loop?

Comment (rphmeier) [inline scripts/aura-test.sh]: pseudocode like this

```
for test in parity-import-tests/*
  cd parity-import-tests/$test
  parity --chain chain.json -d /tmp/import-tests import blocks.rlp
  parity --chain chain.json -d /tmp/import-tests restore snap
  cd ../..
```

Comment (rphmeier) [inline scripts/aura-test.sh]: each directory has 3 files:
  - chain.json
  - blocks.rlp
  - snap

Comment (General-Beck) [inline scripts/aura-test.sh]: @rphmeier Will there be a given amount of tests?

Comment (rphmeier) [inline scripts/aura-test.sh]: no, there is no cap. we will add them periodically, whenever necessary.

**Comment by gavofyork** [comment]
@General-Beck anything happening here?

**Comment by rphmeier** [comment]
this should be revived @paritytech/core-devs. IMO integration tests for varying chains are a pretty critical part of our test infrastructure.

**Pull request closed without merging**